src/dir.props

```diff
@@ -7,4 +7,8 @@
     <SolutionFile Condition="'$(OsEnvironment)'=='Windows_NT'">$([System.IO.Path]::Combine($(SolutionRoot),"MSBuild.sln"))</SolutionFile>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(EnableDocumentationFile)' == 'true'">
+    <DocumentationFile>$(OutputPath)$(TargetName).xml</DocumentationFile>
```

Comment:
The DocumentationFile property must be set before importing the targets

[https://github.com/Microsoft/msbuild/blob/master/src/XMakeTasks/Microsoft.CSharp.CurrentVersion.targets#L155](https://github.com/Microsoft/msbuild/blob/master/src/XMakeTasks/Microsoft.CSharp.CurrentVersion.targets#L155)

I guess this could be inside the dirs.targets before it imports Microsoft.CSharp.targets.  Is that preferable?

---
In reply to: [78440641](https://github.com/Microsoft/msbuild/pull/1015#discussion_r78440641) [](ancestors = 78440641)